src/reporter.cc

```diff
@@ -71,48 +82,134 @@ void MetadataReporter::ReportMetadata() {
 namespace {
 
 void SendMetadataRequest(std::vector<json::value>&& entries,
-                         const std::string& endpoint,
+                         const std::string& host,
+                         const std::string& publish_endpoint,
                          const std::string& auth_header,
                          const std::string& user_agent,
-                         bool verbose_logging)
+                         bool verbose_logging,
+                         bool use_batch)
     throw (boost::system::system_error) {
-  json::value update_metadata_request = json::object({
-    {"entries", json::array(std::move(entries))},
-  });
 
-  if (verbose_logging) {
-    LOG(INFO) << "About to send request: POST " << endpoint
-              << " User-Agent: " << user_agent
-              << " " << *update_metadata_request;
-  }
+  if (use_batch) {
```

Comment:
Why would you ever not use batch?

Reply:
I added the switch since the batch endpoint was not ready. Even now, it is not quite working yet.